common/lcs/src/ser.rs

```diff
@@ -91,22 +91,26 @@ where
 
 pub fn is_human_readable() -> bool {
     let mut output = Vec::new();
-    let serializer = Serializer::new(&mut output);
+    let serializer = Serializer::new(&mut output, crate::MAX_OBJECT_DEPTH);
     ser::Serializer::is_human_readable(&serializer)
 }
 
 /// Serialization implementation for LCS
 struct Serializer<'a, W> {
     output: &'a mut W,
+    max_remaining_depth: usize,
 }
 
 impl<'a, W> Serializer<'a, W>
 where
     W: std::io::Write,
 {
     /// Creates a new `Serializer` which will emit LCS.
-    fn new(output: &'a mut W) -> Self {
-        Self { output }
+    fn new(output: &'a mut W, max_remaining_depth: usize) -> Self {
```

Comment:
I suspect you want to check that max_remaining_depth > 0.

Reply:
This function is not `pub`. Also, `0` could actually make sense for testing.